### DISCUSSION TITLE
Rowy advanced filtering

### DISCUSSION THREAD

**Discussion by ItaliaNorway** [AntlerVC/firetable#1476] [Support / Q&A] [answered]
Hi, my name is Riccardo and I´m a product designer. 
I`m developing an app: Luup app an "used items" exchange platfrom. 
The users take pictures and videos of the items the want to exchange, set a value, upload the item in the platform. 
Users can view an items gallery and filter it by category, material, size, value, etc 
Select the item they want to "buy" . Pay for the delivery (using stripe), send a request to the item owner. 
The owner send the item. 
I have sucessfully created a full workign prototype usign Figma, Bravo Studio, Firebase, Airtable and Xano. I have integrated delivery, payment (stripe), and notification (oneSignal)  API´s ...
I used airtable and Xano because they are no-code, low-code easy to use and well documented in Bravo Studio documentations.

Since I´m almost ready for deployment (and I found Rowy) I tought was better to "migrate" everything in one platform (firebase), but I´m regretting this choice :( 

I started migrating the "items" table from airtable to firestore using Rowy. (is video supported?)  and I wanted to add the complex filtering I was able to create with a Xano functions quering Airtable...

here the mess started. I realised firestore do not support any "complex" filtering so I installed algolia but also here I have problems.
 
bravo studio will send a URL request with boolean variables that express the user filtering selection (ex: if Adult_clothes=true, women=true, sizeS=true and sizeM=true) I want to return only women adult clothes in size S and M ...
Now I created a firebase function that take this Url request, ask algoria to filter the firestore database and return the filtered results.
it works but I saw that now Rowy supports complex filtering can I use this instead? How I connect it to bravo studio?
Anyone can help?

**Comment by ItaliaNorway** [comment]
Anyone?

**Reply by Chadha93** [reply]
cc @il3ven Could you help here?

**Comment by bofeiw** [comment, marked as answer]
Let me simplify the issue - you need 1. complex (or multiple) filters in the Rowy UI, 2. image/video support.

With the [Rowy 3.0](https://github.com/rowyio/rowy/releases/tag/v3.0.0) release, multiple filters are supported. We also have [Full text search](https://docs.rowy.io/full-text-search) in the table UI as an optional choice.

However, based on your description, you might want to implement [Webhooks](https://docs.rowy.io/webhooks) to handle data sent from Bravo Studio. In the Webhooks, you can write code to achieve any conditions for the filters you need. However, I would recommend you to try [BuildShip](https://buildship.com), built by the Rowy team, it lets you create endpoints fast with native integration with Rowy.

Images and videos are supported. Rowy has native [Image/file column](https://docs.rowy.io/field-types/image-file) for you to use. You can drag and drop to upload to your Firebase Storage bucket.
